packages/agent/src/connect-v2.ts

```diff
@@ -0,0 +1,199 @@
+import { utils } from '@web5/crypto';
+import { DwnProtocolDefinition, DwnRecordsPermissionScope } from './index.js';
+import {
+  HybridAuthResponse,
+  Oidc,
+  type PushedAuthResponse,
+} from './oidc-v2.js';
+import { pollWithTTL } from './utils.js';
+import { Convert } from '@web5/common';
+import { DidDht } from '@web5/dids';
+
+async function initClient({
+  connectServerUrl,
+  permissionRequests,
+  clientUri,
+  onUriReady,
+  validatePin,
+}: WalletConnectOptions) {
+  // ephemeral client did for ECDH, signing, verification
+  // TODO: use separate keys for ECDH vs. sign/verify. could maybe use secp256k1.
+  const clientDid = await DidDht.create();
+
+  // Hash the code verifier to use as a code challenge and to encrypt the Request Object.
+  const { codeVerifieru8a } = Oidc.generateRandomCodeVerifier();
+
+  // Derive the code challenge based on the code verifier
+  const { codeChallengeu8a, codeChallengeb64url } =
```

Comment:
Naming convention (`camelCasing`) and bad naming (`u8a`).

```suggestion
  const { codeChallengeBytes, codeChallengeBase64Url } =
```

Reply:
agree on bytes but isn't the convention for base64url `b64url`?

Reply:
Did a string search in the repo, "base64url" occurred 400+ times, while b64url only showed up in this PR, so "Base64Url" is the convention.

Reply:
done